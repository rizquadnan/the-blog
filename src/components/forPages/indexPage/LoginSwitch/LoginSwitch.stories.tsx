import * as React from 'react'
import { Meta, Story } from '@storybook/react'
import { LoginSwitch } from './LoginSwitch'

type TLoginSwitch = React.ComponentProps<typeof LoginSwitch>
export default {
  title: 'UserInputs/LoginSwitch',
  component: LoginSwitch,
} as Meta<TLoginSwitch>

const Template: Story<TLoginSwitch> = (args) => <LoginSwitch {...args} />

export const Default = Template.bind({})
