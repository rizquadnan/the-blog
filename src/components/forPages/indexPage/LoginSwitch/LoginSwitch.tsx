import React from 'react'
import { Container, Heading, Text } from '@chakra-ui/layout'
import { Button } from '@chakra-ui/button'
import { VStack } from '@/components/uiKit/VStack'
import { Link } from '@/components/uiKit/Link'

type TLoginSwitch = {
  onClickGuestMode: () => void
  onClickAdmin: () => void
  onClickUser: () => void
}
export function LoginSwitch(props: TLoginSwitch) {
  return (
    <Container
      maxW="container.sm"
      borderColor="white"
      border="1px solid"
      borderRadius="4px"
      padding="32px"
    >
      <VStack textAlign="center" spacing="32px">
        <VStack spacing="4px">
          <Heading as="h1">Welcome to The Blog</Heading>
          <Text>
            You can tryout this app in guest mode! For complete features, login
            using admin account or a user account
          </Text>
        </VStack>
        <VStack w="100%">
          <Button w="100%" as={Link} href="/posts" onClick={props.onClickAdmin}>
            Guest Mode
          </Button>
          <Button w="100%" as={Link} href="/posts" onClick={props.onClickAdmin}>
            Login As Admin
          </Button>
          <Button w="100%" as={Link} href="/posts" onClick={props.onClickUser}>
            Guest As User
          </Button>
        </VStack>
      </VStack>
    </Container>
  )
}

export default LoginSwitch
