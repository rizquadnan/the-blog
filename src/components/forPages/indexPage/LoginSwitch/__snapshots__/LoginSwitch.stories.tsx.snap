// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Inputs/LoginSwitch Default smoke-test 1`] = `
Object {
  "checked": undefined,
  "children": Array [
    Object {
      "checked": undefined,
      "children": undefined,
      "level": 1,
      "name": "Welcome to The Blog",
      "pressed": undefined,
      "role": "heading",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "You can tryout this app in guest mode! For complete features, login using admin account or a user account",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Guest Mode",
      "pressed": undefined,
      "role": "button",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Login As Admin",
      "pressed": undefined,
      "role": "button",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Guest As User",
      "pressed": undefined,
      "role": "button",
      "value": undefined,
    },
  ],
  "name": "user-inputs-loginswitch--default",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;
